rst_files/generic_programming.rst

```diff
@@ -713,12 +724,15 @@ You will notice that types in Julia represent a tree with ``Any`` at the root
 
 The tree structure has worked well for the above examples, but it doesn't allow us to associate multiple categorizations of types
 
-For example, a semi-group type would be useful for a writing generic code (e.g. continuous-time solutions for ODEs and matrix-free methods), but cannot be implemented rigorously since the ``Matrix`` type is a semi-group as well as an ``AbstractArray``, but not not all ``semi-groups" are ``AbstractArray``
+For example, a semi-group type would be useful for a writing generic code (e.g. 
+continuous-time solutions for ODEs and matrix-free methods), but cannot be 
+implemented rigorously since the ``Matrix`` type is a semi-group as well 
+as an ``AbstractArray``, but not all ``semi-groups`` are ``AbstractArray``
```

Comment:
The problem is that ``semi-group`` is not a defined structure in the libraries we use, so we should instead change it to either "semi-group" or perhaps just drop the quotes around semi-group

Reply:
Dropped the backticks.